File: lib/src/main/java/com/sd/lib/cache/handler/impl/FloatHandler.java
package com.sd.lib.cache.handler.impl;

import com.sd.lib.cache.CacheInfo;
import com.sd.lib.cache.handler.BaseCacheHandler;

/**
 * Float处理类
 */
public class FloatHandler extends BaseCacheHandler<Float> {
    public FloatHandler(CacheInfo info) {
        super(info);
    }

    @Override
    protected byte[] valueToByte(Float value) {
        return value.toString().getBytes();
    }

    @Override
    protected Float byteToValue(byte[] bytes, Class<?> clazz) {
        return Float.valueOf(new String(bytes));
    }

    @Override
    protected String getKeyPrefix() {
        return "float_";
    }
}

File: lib/src/main/java/com/sd/lib/cache/handler/impl/FloatHandler.kt
package com.sd.lib.cache.handler.impl

import com.sd.lib.cache.CacheInfo
import com.sd.lib.cache.handler.BaseCacheHandler

/**
 * Float处理类
 */
internal class FloatHandler(info: CacheInfo) : BaseCacheHandler<Float>(info, "float") {
    override fun encodeToByteImpl(value: Float, clazz: Class<*>?): ByteArray {
        return value.toString().toByteArray()
    }

    override fun decodeFromByteImpl(bytes: ByteArray, clazz: Class<*>?): Float {
        return String(bytes).toFloat()
    }
}